django_setup_configuration/config_settings.py

```diff
@@ -0,0 +1,227 @@
+        if choices := field.choices:
+            example_values = [choice[0] for choice in choices]
+            return ", ".join(example_values)
+
+        # other fields
+        field_type = type(field)
+        if field_type in basic_field_descriptions.keys():
+            return basic_field_descriptions.get(field_type)
+
+        return "No information available"
+
+    def create_config_fields(
+        self,
+        model: Type[models.Model],
+        relating_field: models.Field | None = None,
+    ) -> None:
+        """
+        Create a `ConfigField` instance for each field of the given `model` and
+        add it to `self.fields`
+
+        Basic fields (`CharField`, `IntegerField` etc) constitute the base case,
+        relations (`ForeignKey`, `OneToOneField`) are handled recursively
+        """
+
+        for model_field in model._meta.fields:
+            if isinstance(model_field, (ForeignKey, OneToOneField)):
+                # avoid recursion error when following ForeignKey
+                if model_field.name in ("parent", "owner"):
+                    continue
+
+                self.create_config_fields(
+                    model=model_field.related_model,
+                    relating_field=model_field,
+                )
+            else:
+                # model field name could be "api_root",
+                # but we need "xyz_service_api_root" (or similar) for consistency
+                # when dealing with relations
+                if relating_field:
+                    config_field_name = f"{relating_field.name}_{model_field.name}"
+                else:
+                    config_field_name = model_field.name
+
+                config_setting = self.get_config_variable(config_field_name)
+
+                if not (
+                    config_setting in self.required_settings
+                    or config_setting in self.optional_settings
+                ):
+                    continue
+
+                config_field = ConfigField(
+                    name=config_field_name,
+                    verbose_name=model_field.verbose_name,
+                    description=model_field.help_text,
+                    default_value=self.get_default_value(model_field),
+                    field_description=self.get_field_description(model_field),
+                )
+                self.config_fields.append(config_field)
+
+    #
+    # convenience methods/properties for formatting
+    #
+    def get_config_variable(self, setting: str) -> str:
+        return f"{self.namespace}_" + setting.upper()
+
+    @property
+    def file_name(self) -> str:
+        """
+        Use `self.namespace` in lower case as default file name of the documentation
+        if `file_name` is not provided when instantiating the class
+        """
+        return getattr(self, "_file_name", None) or self.namespace.lower()
+
+    @file_name.setter
+    def file_name(self, val) -> None:
+        self._file_name = val
```

Comment:
Is there a reason to use this construction here? Can't you just make the `file_name` argument optional in the init and set it to `self.namespace.lower()` if it's not supplied?